doc/syslog/python-logger-enhancement.md

```diff
@@ -0,0 +1,217 @@
+children = []
+
+def child_signal_handler(signum, frame):
+    syslogger.SysLogger.refresh_config()
+
+def child():
+    signal.signal(signal.SIGHUP, child_signal_handler)
+    child_log = syslogger.SysLogger('daemon_name', enable_runtime_config=True)
+    while True:
+        child_log.log_notice('log something')
+
+def main_signal_handler(signum, frame):
+    global children
+    if signum == signal.SIGHUP:
+        syslogger.SysLogger.refresh_config()
+        for child in children:
+            os.kill(child.pid, signal.SIGHUP)
+
+signal.signal(signal.SIGHUP, main_signal_handler)
+p = multiprocessing.Process(target=child)
+children.append(p)
+p.join()
+```
+
+### SAI API
+
+N/A
+
+### Configuration and management
+
+#### Manifest (if the feature is an Application Extension)
+
+N/A
+
+#### CLI/YANG model Enhancements
+
+Currently, `swssloglevel` is used to update log level at run time. It relies on the thread running in `swsscommon.Logger` to refresh the log level of each logger instance. Since there is no such thread in python logger implementation, a new CLI shall be added to refresh the log level by sending signal to relevant daemon.
+
+> Note: `swssloglevel` cannot send signal to daemons running in other container because it is actually a program installed in swss container.
+
+config syslog level -c <component> -l <log_level> [--service <service_name>] [--program <program_name>] [--pid <pid>]
+
+- "-c": component name or log identifier name. It is the key of LOGGER table.
+- "-l": log level. One of "DEBUG", "INFO", "NOTICE", "WARN", "ERROR"
+- "--service": container name. Optional. The signal will be sent to that container if present.
+- "--program": program name. Optional, only applicable when --service is present. The signal will be sent to that program running in the service.
+- "--pid": process ID. Optional. If --service is present, the signal will be sent to the PID running in the service; otherwise, the signal will be send to the PID running in host side.
+
+Examples:
+```
+# Update the log level without refresh the configuration
+config syslog level -c xcvrd -l DEBUG
+
+# Update the log level and send SIGHUP to xcvrd running in PMON
+config syslog level -c xcvrd -l DEBUG --service pmon --program xcvrd
+
+# Update the log level and send SIGHUP to PID 20 running in PMON
+config syslog level -c xcvrd -l DEBUG --service pmon --pid 20
+
+# Update the log level and send SIGHUP to PID 20 running in host
+config syslog level -c xcvrd -l DEBUG --pid 20
+
+# Send SIGHUP to xcvrd running in PMON without update log level
+config syslog level --service pmon --program xcvrd
+
+# Send SIGHUP to PID 20 running in PMON without update log level
+config syslog level --service pmon --pid 20
+
+# Send SIGHUP to PID 20 running in host without update log level
+config syslog level --pid 20
+
+# Invalid, --program must be used together with --service
+config syslog level --program xcvrd
+
```

Comment:
@Junchao-Mellanox can we have option `signal` instead of `level` to send signal?

Reply:
Hi @prgeor , the requirement is to have one single command which update log level and make it effect, so, I have to add these two function together.